app/src/test/kotlin/compiler/intermediate_form/ExpressionControlFlowTest.kt

```diff
@@ -1,12 +1,673 @@
+                }
+            }
+        }
+
+        fun createCfg(expr: Expression, targetVariable: Variable? = null): ControlFlowGraph {
+            return ControlFlow.createGraphForExpression(expr, targetVariable, currentFunction, nameResolution, variableProperties, finalCallGraph, functionDetailsGenerators, argumentResolution)
+        }
+    }
+
+    private infix fun String.asVarIn(exprContext: ExpressionContext): Variable {
+        return exprContext.nameToVarMap[this]!!
+    }
+
+    private infix fun String.asFunIn(exprContext: ExpressionContext): Function {
+        return exprContext.nameToFunMap[this]!!
+    }
+
+    private infix fun Pair<String, List<Expression>>.asFunCallIn(exprContext: ExpressionContext): Expression.FunctionCall {
+        val result = Expression.FunctionCall(this.first, this.second.map { Expression.FunctionCall.Argument(null, it) })
+        for (i in 0 until result.arguments.size) {
+            exprContext.argumentResolution[result.arguments[i]] = (this.first asFunIn exprContext).parameters[i]
+        }
+        exprContext.nameResolution[result] = exprContext.nameToFunMap[this.first]!!
+        return result
+    }
+
+    private infix fun String.asFunCallIn(exprContext: ExpressionContext): Expression.FunctionCall {
+        return Pair<String, List<Expression>>(this, emptyList()) asFunCallIn exprContext
+    }
+
+    private infix fun String.withArgs(list: List<Expression>): Pair<String, List<Expression>> {
+        return Pair(this, list)
+    }
+
+    private infix fun String.asVarExprIn(exprContext: ExpressionContext): Expression.Variable {
+        val result = Expression.Variable(this)
+        exprContext.nameResolution[result] = this asVarIn exprContext
+        return result
+    }
+
+    private infix fun Expression.add(expr: Expression): Expression {
+        return Expression.BinaryOperation(Expression.BinaryOperation.Kind.ADD, this, expr)
+    }
+
+    private infix fun Expression.and(expr: Expression): Expression {
+        return Expression.BinaryOperation(Expression.BinaryOperation.Kind.AND, this, expr)
+    }
+
+    private infix fun Expression.or(expr: Expression): Expression {
+        return Expression.BinaryOperation(Expression.BinaryOperation.Kind.OR, this, expr)
+    }
+
+    private fun ternary(cond: Expression, ifTrue: Expression, ifFalse: Expression): Expression {
+        return Expression.Conditional(cond, ifTrue, ifFalse)
+    }
+
+    private infix fun ControlFlowGraph.hasSameStructureAs(cfg: ControlFlowGraph): Boolean {
+        val registersMap = ReferenceHashMap<Register, Register>()
+        val callResultsMap = ReferenceHashMap<IntermediateFormTreeNode.DummyCallResult, IntermediateFormTreeNode.DummyCallResult>()
+        val nodeMap = ReferenceHashMap<IntermediateFormTreeNode, IntermediateFormTreeNode>()
+
+        fun <T> ReferenceHashMap<T, T>.isBijection(a: T, b: T): Boolean {
```

Comment:
```suggestion
        fun <T> ReferenceHashMap<T, T>.ensurePairSymmetrical(a: T, b: T): Boolean {
```
The previous name suggests this function has no side-effects